pkg/sql/schemachanger/scbuild/internal/scbuildstmt/zone_config_helpers.go

```diff
@@ -1395,12 +1399,15 @@ func configureZoneConfigForNewIndexBackfill(
 	newSubzones = append(newSubzones, newZoneConfig.Subzones...)
 	// For the indexes we will use as a part of the backfill, ensure we copy
 	// over each subzone config from the old index to the backfill-related ones.
+	// NOTE: The subzones for the old index and temporary index will eventually
+	// be removed by the schema change GC job, but we need them to be present
+	// for the duration of this schema change.
 	for _, idxToAdd := range newIndexesForBackfill {
 		for _, subzone := range newZoneConfig.Subzones {
 			if subzone.IndexID == uint32(oldIndexID) {
 				subzone.IndexID = uint32(idxToAdd)
+				newSubzones = append(newSubzones, subzone)
```

Comment:
ah one thing: this will also now erase any currently existing subzones we want to keep, right? like if we had secondary indexes with subzone configs, but were doing a backfill on the primary index, this would not capture those subzone configs

Reply:
actually ignore me: this is why the bug exists in the first place -- i forget we copy over all the subzones first in `newSubzones`